Issue 3188: Using Lombok Getter(lazy=true) in AbstractScaleTests.java make javac crash on JDK10

eolivelli opened the pull request:
**Change log description**  
- Fix compilation error in AbstractScaleTests.java due to Lombok Getter(lazy=true) on JDK10

**Purpose of the change**  
Make the build pass in JDK10
Fixes #3188

eolivelli commented:
@fpj  @andreipaduroiu @tkaitchuck 

This change is the last in order to make the build pass on JDK10, all tests passed on Travis.
I can split the patch in three distinct patches (spotbugs + Getter issue + Travis) if you prefer.

tkaitchuck commented:
I believe I have checked in a PR which addresses these issues: https://github.com/pravega/pravega/commit/b80b6272509fd8a6f8992c54b2eb78871201d26d
Does this solve the problem?

eolivelli commented:
@tkaitchuck are you referring to the spotbug errors or to Getter(lazy=true) ?

For the spotbugs errors the only workaround is to suppress the errors.
For lombok stuff I can try to rebuild on jdk without dropping 'lazy'.

Btw that 'lazy' attibutes is not really useful in this case and having an explicit call to init code seems to me clearer.

eolivelli commented:
@tkaitchuck without this fix on current master I have this javac error while building :test:system:test on jdk10

```
An exception has occurred in the compiler (10.0.2). Please file a bug against the Java compiler via the Java bug reporting page (http://bugreport.java.com) after checking the Bug Database (http://bugs.java.com) for duplicates. Include your program and the following diagnostic in your report. Thank you.
java.lang.NullPointerException
        at jdk.compiler/com.sun.tools.javac.jvm.Code.typecode(Code.java:231)
        at jdk.compiler/com.sun.tools.javac.jvm.Items$Item.coerce(Items.java:269)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.genExpr(Gen.java:819)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.genArgs(Gen.java:836)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.visitNewClass(Gen.java:1737)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCNewClass.accept(JCTree.java:1689)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.genExpr(Gen.java:817)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.genArgs(Gen.java:836)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.visitNewClass(Gen.java:1737)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCNewClass.accept(JCTree.java:1689)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.genExpr(Gen.java:817)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.visitVarDef(Gen.java:1015)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCVariableDecl.accept(JCTree.java:956)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.genDef(Gen.java:595)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.genStat(Gen.java:630)
        at jdk.compiler/com.sun.tools.javac.jvm.Gen.genStat(Gen.java:616)

```

eolivelli commented:
@tkaitchuck 
spotbugs errors are about this incompatibility of spotbugs with JDK9-10-11 https://github.com/spotbugs/spotbugs/issues/432

tkaitchuck commented:
Yes, I ask because I also tried building with Java 10 and before that PR I saw that same exception and now I do not.

eolivelli commented:
Okay. Will do

eolivelli commented:
@tkaitchuck I have reverted the change about spotbugs configuration

eolivelli commented:
Ping

eolivelli commented:
Kindly pinging
@tkaitchuck  @andreipaduroiu  @fpj

fpj commented on .travis.yml:
```diff
@@ -11,6 +11,7 @@ language: java
 install: true
 jdk:
  - oraclejdk8
+ - openjdk10
```
I'm not sure why we need to install `openjdk10` in the travis configuration for this PR. If you simply wanted to test, then can we remove it for merging?

eolivelli commented:
@fpj  I have reverted the change to Travis configuration
I had added it in order not to make changes which break builds on JDK10, as the team is currently using only JDK8 (AFAIK).

I have changed the description as @tkaitchuck pointed that spotbugs errors where fixed with another commit, and while addressing his comments I had reverted that change.

Now the patch is only about AbstractScaleTests.java

thank you
I will proceed with JDK11
